agentops/__init__.py

```diff
@@ -79,14 +81,20 @@ def init(
         skip_auto_end_session=skip_auto_end_session,
     )
 
-    return inherited_session_id or c.current_session_id
+    if inherited_session_id:
+        return inherited_session_id
+    elif len(c.current_session_ids) is 1:
+        return c.current_session_ids[0]
+    else:
+        return None
 
 
 def end_session(
```

Comment:
Not saying this has to be done here, but a session context manager could be a cool future addition